### ISSUE TITLE
Safe reflection, re-exports builtins, and more

### ISSUE THREAD

**Issue by L-TChen** [agda/agda-stdlib#698]
There are few issues about `Reflection` module 

* It imports a number of unsafe modules, but the `Agda.Builtin.Reflection` comes with `--safe` enabled by default. I wonder if it is better to have a submodule which collects all unsafe features. 

* Some built-ins are not re-exported by `Reflection`, e.g., `Associativity`, `Precedence`, and `Fixity`. 

* I think it would be nice to have some pattern synonyms for, e.g., visible relevant argument to ease debugging, e.g.,  
```
pattern vArg ty = arg (arg-info visible relevant) ty
pattern hArg ty = arg (arg-info hidden relevant) ty
pattern vLam s t = lam visible (abs s t)
pattern hLam s t = lam hidden (abs s t)
pattern vΠ⦅_∶_⦆_ s ty₁ ty₂ = pi (vArg ty₁) (abs s ty₂)
pattern hΠ⦅_∶_⦆_ s ty₁ ty₂ = pi (hArg ty₁) (abs s ty₂)
```
Perhaps it is just a matter of taste, but at least they make information shorter. I can make a PR later if what I wrote above makes sense to the users of `Reflection`.

**Comment by MatthewDaggitt** [comment]
> It imports a number of unsafe modules, but the Agda.Builtin.Reflection comes with --safe enabled by default. I wonder if it is better to have a submodule which collects all unsafe features.

Depends how much of the core functionality is unsafe? If only a little bit then yes definitely. 

> Some built-ins are not re-exported by Reflection, e.g., Associativity, Precedence, and Fixity.

These should be fixed. I think @gallais has already raised some of them in #627.

> I think it would be nice to have some pattern synonyms for, e.g., visible relevant argument to ease debugging, e.g.,

:+1: 

If you'd like to put together a pull request implementing some or all of these, then that'd be great.

**Comment by MatthewDaggitt** [comment]
First two points are fixed via 805c897. Making `Reflection` safe is currently blocked by #726.

**Comment by L-TChen** [comment]
In addition, making `Reflection` safe is also blocked by https://github.com/agda/agda/issues/3736.

**Comment by L-TChen** [comment]
It appears that this issue has been solved by the @gallais and @MatthewDaggitt. Close now.
